contributors/devel/security-release-process.md

```diff
@@ -58,7 +58,37 @@ These steps should be completed within the first 24 hours of Disclosure.
 These steps should be completed within the 1-7 days of Disclosure.
 
 - The Fix Lead and the Fix Team will create a [CVSS](https://www.first.org/cvss/specification-document) using the [CVSS Calculator](https://www.first.org/cvss/calculator/3.0). The Fix Lead makes the final call on the calculated CVSS; it is better to move quickly than make the CVSS prefect.
-- The Fix Team will notify the Fix Lead that work on the fix branch is complete once there are LGTMs on all commits in the private repo from one or more relevant assignees in the relevant OWNERS file.
+- To encourage fast adoption of the security fix, the new patched release should
+  contain only security fixes.
+- To facilitate this goal, the Fix Lead will [create a patch release branch](https://github.com/kubernetes/release/blob/master/docs/branching.md#branching-from-a-tag)
+  in a private fork of the public kubernetes.git repository, branched from the
+  most recent released tag for the release being fixed.
+  - As a hypothetical example, if `v2.4.7` is the latest public patch release for
+   `v2.4`, a new branch will be created in the private fork called
+   `release-2.4.7`, branching at the `v2.4.7` tag. If `HEAD` were to be used
+   instead, the security release might include non-security fixes.
+  - TODO: continuous integration testing should be set up for the branch
+- If a security incident affects multiple supported releases, then patch
+  release branches will be created for all affected releases.
+- Similarly, if a security issue affects `master`, then a private `master`
+  branch will be created, though this will be synced to public `HEAD`, rather
+  than from a particular tag.
+- The Fix Team will submit fixes to the private branch(es). If a fix is only
+  needed for a published release (and not `master`), then it is fine to merge
+  directly into the private release branch (rather than following the
+  cherry-pick process).
+  - As is feasible, the normal PR review process (including LGTM and OWNERS
+    approval) should be followed for all fixes, though the submit queue
+    infrastructure may not be operational on the private branches.
+- The Fix Team will notify the Fix Lead that work on the fix branch(es) is
+  complete once all relevant changes have been merged into the private
+  branch(es).
+- Release Managers are encouraged to refrain from creating new public patch
+  releases while security fixes are ongoing. Should it be necessary to create a
+  new public patch release before the security fixes are complete, the private
+  security branch will need to be rebased (and renamed?) on the new latest
+  patch release.
```

Comment:
You raise a good point.

I think we should only build a single security patch release for each issue - so if we branch from `v2.4.7`, and `v2.4.8` is released without a fix, we ship only `v2.4.9` with the fix (after rebasing on `v2.4.8`). I would guess that the vulnerability would've existed in all `v2.4` patch releases, and we certainly don't want to build and release a `v2.4.0-r2`, `v2.4.1-r2`, etc. (I guess this raises questions about support commitments - do we support users at every patch release for a given X.Y release? Or only the latest patch release? We certainly don't regularly test anything besides latest patch release.)

The challenge here is that delaying the non-security patch release may reveal the existence of a vulnerability, though that's probably OK, especially if the disclosure of fix (detailed in the next section) has already occurred.

If there is a serious bug affecting production clusters which needs the non-security patch immediately, then this is a reasonable case to break the "no non-security fixes" rule.
In such cases, we'd probably need to send a follow-up to the original disclosure email, noting that the public patch release doesn't include the fix, but recommending an upgrade to make the security upgrade go as smoothly as possible.